### ISSUE TITLE
Unable to find Qt

### ISSUE THREAD

**Issue by Shatur95** [tonka3000/vscode-qt-tools#26]
Hello! This extension cannot detect Qt. But I have it installed and I can build Qt applications (I also have `vscode-cmake-tools` installed). I am using Arch Linux and QtCreator with other tools just available from `PATH`. Could you add automatic search in system `PATH`?

**Comment by tonka3000** [comment]
Hey,

* Do you use the cmake commands from the cmake extension (configure), correct?
* Do you use cmake from Qt creator before?

My extension extract the information of the Qt files from cmake based projects only. If you are using cmake please provide the command how you installed Qt on your system and also upload your CMakeCache file from your binary directory here so that I can reproduce your environment. It could be that your directory layout of your Qt installation is different on your system.

Greetings

Tonka

**Comment by Shatur95** [comment]
> Do you use the cmake commands from the cmake extension (configure), correct?
> Do you use cmake from Qt creator before?

Yep. Also Qt project compiles and works from QtCreator and VSCode.

I used my package manager to install Qt for my distribution (from the official repository). For example, here is the [base](https://www.archlinux.org/packages/extra/x86_64/qt5-base/) package (scroll down and click to `View the file list for qt5-base`). I am using KDE, so I have all the necessary Qt packages installed (Arch Linux does not split into devel packages).

[CMakeCache.txt](https://github.com/tonka3000/vscode-qt-tools/files/5059242/CMakeCache.txt)

**Comment by tonka3000** [comment]
Interesting. There is no reference in your cache file which points to f.e. moc executable which is unusual because qt mostly store the found variables in the cache. I have to check the qt cmake files to understand how they search for the files because I want to avoid finding qt version which is different from the version cmake finds. I will check it in the next couple of days.

Greetings

Tonka

**Comment by Shatur95** [comment]
You need this variables to run Qt tools?
Usually it system `PATH` or [Qt5_DIR](https://github.com/jurplel/install-qt-action#more-info) variables. This will also work for non-CMake projects.

**Comment by tonka3000** [comment]
> You need this variables to run Qt tools?
Yes. In the actual impl..

> Usually it system PATH or Qt5_DIR variables. This will also work for non-CMake projects.
Correct. On Windows then can be a lit bit tricky because often I have the something in the PATH and `Qt5_DIR`.

**Comment by Shatur95** [comment]
I have a suggestion. Add the ability to specify paths to Qt Tools manually (array, to let user set several paths). If Qt was not found, check `PATH` and `Qt5_DIR`. If Qt still not found - try to get it from CMake.

**Comment by tonka3000** [comment]
You can already set `qttools.extraSearchDirectories` which is an array of lookup paths, so that Qt designer and assistant will be searched there.

**Comment by tonka3000** [comment]
Hey @Shatur95 ,

please can you build the #27 branch yourself and check if it fix your problem.

Thanks in advance

Tonka
